kafka/config_sasl.go

```diff
@@ -18,7 +18,8 @@ type SASLConfig struct {
 	Mechanism string `koanf:"mechanism"`
 
 	// SASL Mechanisms that require more configuration than username & password
-	GSSAPI SASLGSSAPIConfig `koanf:"gssapi"`
+	GSSAPI      SASLGSSAPIConfig  `koanf:"gssapi"`
+	OAuthBearer OAuthBearerConfig `koanf:"oauthbearer"`
```

Comment:
```suggestion
	OAuthBearer OAuthBearerConfig `koanf:"oauth"`
```

Primarily to align with the config that is used in Redpanda Console. Also probably less error-prone when users try to use `oauthbearer` or `oauthBearer`